route-snapper/src/lib.rs

```diff
@@ -1024,10 +1025,15 @@ impl Router {
         // back
         // TODO Seems fast enough, but we could cache and build this up incrementally
         let mut avoid = HashSet::new();
+        let mut prev_edge_name = None;
+
         if self.config.avoid_doubling_back {
             for entry in prev_path {
                 if let PathEntry::Edge(e) = entry {
                     avoid.insert(e.0);
+                    if self.config.match_last_road {
```

Comment:
The path could be quite long and this potentially makes a bunch of unnecessary clones. Maybe we could do something like `prev_path.last()` or avoid the clone and do `map.edge(e).name.as_ref()` or something